### PR TITLE
DDF-3941 - Moved intrigue from using the PlatformUiConfiguration Endpoint under /services

### PR DESCRIPTION
**Pull request by blen-desta**
#### What does this PR do?
**This is PR 2/7**
- Moved the `/services/platform/config/ui` endpoint to PlatformUiConfigurationEndpoint
- Implementation of endpoint remains at PlatformUiConfiguration
- Created new endpoint under `/search/catalog/internal/platform/config/ui` using the same implementation
- Changed `catalog-search-ui` code to use new endpoint

#### Who is reviewing it? 
@garrettfreibott @brjeter @bakejeyner 

#### Ask 2 committers to review/merge the PR and tag them here. If you don't know who to ask, you can request reviews in https://groups.google.com/forum/#!forum/ddf-developers .
(please choose ONLY two committers from below, delete the rest)
@andrewkfiedler 
@coyotesqrl

#### How should this be tested? (List steps with links to updated documentation)
- Make sure the configurations like the header, footer, color etc.. work on intrigue
- Make sure from the network tab that the call is made to the `/search/catalog/internal/platform/config/ui` endpoint
- Change the platform configurations in the Admin Console under `Platform -> Configuration -> Platform UI Configuration` and make sure they take effect in both the Admin Console and Intrigue

#### What are the relevant tickets?
[DDF-3941](https://codice.atlassian.net/browse/DDF-3941)

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: platform/platform-configuration-service/src/main/java/org/codice/ddf/configuration/service/PlatformUiConfigurationService.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.configuration.service;
+
+public interface PlatformUiConfigurationService {
+
+  String getConfig();
```
Comment (brjeter) [inline platform/platform-configuration-service/src/main/java/org/codice/ddf/configuration/service/PlatformUiConfigurationService.java]: ✏️ This method is difficult to understand without going to the impl. How is the returned String formatted? I'd recommend either 1) Adding javadoc or 2) Renaming this method to something like `getConfigAsJsonString`. Or maybe both.

File: platform/platform-configuration/pom.xml [hunk 2]
```diff
@@ -83,6 +83,11 @@
             <version>${project.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>ddf.platform</groupId>
+            <artifactId>platform-configuration-service</artifactId>
+            <version>2.13.0-SNAPSHOT</version>
```
Comment (brjeter) [inline platform/platform-configuration/pom.xml]: ❗️ This should be `${project.version}`.

File: platform/platform-configuration-service/pom.xml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+ **/
+ -->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <parent>
+        <groupId>ddf.platform</groupId>
+        <artifactId>platform</artifactId>
+        <version>2.13.0-SNAPSHOT</version>
```
Comment (brjeter) [inline platform/platform-configuration-service/pom.xml]: ❗️ This should be `${project.version}`.

File: platform/platform-configuration-endpoint/pom.xml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+ -->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>ddf.platform</groupId>
+        <artifactId>platform</artifactId>
+        <version>2.13.0-SNAPSHOT</version>
```
Comment (brjeter) [inline platform/platform-configuration-endpoint/pom.xml]: ❗️ This should be `${project.version}`.

Comment (blen-desta) [reply platform/platform-configuration-endpoint/pom.xml]: This is also on the parent.

**Comment by blen-desta** [comment]
build now

**Comment by blen-desta** [comment]
build now

**Comment by blen-desta** [comment]
build now

**Comment by blen-desta** [comment]
`Failed to execute goal on project catalog-core-solrcommands: Could not resolve dependencies for project ddf.catalog.core:catalog-core-solrcommands:bundle:2.13.0-SNAPSHOT: Could not transfer artifact org.apache.ant:ant:jar:1.8.2 from/to office `
build now

**Comment by blen-desta** [comment]
```
05:22:56 Failed tests: 
05:22:56   ITInstallProfilesFeatures.installAndUninstallFeature:79 The bundle DDF :: Catalog :: Solr :: Solr REST Client failed.
05:22:56   ITInstallProfilesFeatures.installAndUninstallFeature:79 The bundle DDF :: Catalog :: Solr :: Solr REST Client failed.
```
Unrelated failure
build now

**Comment by blen-desta** [comment]
`[ERROR] One or more dependencies were identified with vulnerabilities that have a CVSS score greater than or equal to '7.0':
17:56:37 [ERROR] 
17:56:37 [ERROR] spring-core-4.3.10.RELEASE.jar: CVE-2018-1275`

**Comment by clockard** [comment]
build now

**Comment by blen-desta** [comment]
Failure:
```
Failed to execute goal org.owasp:dependency-check-maven:3.1.1:check (default) 
on project platform-util:
03:44:28 [ERROR] 
03:44:28 [ERROR] One or more dependencies were identified with vulnerabilities that have 
a CVSS score greater than or equal to '7.0':
03:44:28 [ERROR] 
03:44:28 [ERROR] spring-core-4.3.10.RELEASE.jar: CVE-2018-1275
```

**Comment by clockard** [comment]
Bypassed CI because it only failed for an unrelated owasp issue that is getting fixed in a different pr.

**Pull request merged**